docs/tutorial/algorithm_interfaces.rst

```diff
@@ -1,2 +1,464 @@
 Algorithm interfaces
-====================
\ No newline at end of file
+====================
+
+In the previous chapter, we went over the :ref:`big picture of ALF <chap2_big_picture>`.
+Now we will take a look at how different algorithm interfaces interact with each
+other in the on-policy/off-policy pipelines. We will try modifying some
+algorithm interfaces of the previous example :doc:`./a_minimal_working_example`,
+for an illustration purpose. Then we introduce a simple off-policy training
+example and again explore its algorithm interfaces.
+
+Interface Overview
+------------------
+
+.. image:: images/algorithm_interfaces.png
+    :width: 800
+    :align: center
+    :alt: interaction between algorithm interfaces
+
+
+There are several key interfaces (highlighted in the figure above) that play
+important roles in assembling a training pipeline of an ALF algorithm, and their
+signatures are summarized below.
+
+======================================================  ===========================================  =======================================  =============================== ===============
+Interface                                               Tensor shape                                 Input                                    Output                          Which pipeline?
```

Comment:
Might be clearer if we can explain whether the `Tensor Shape` here in the table stands for the input, the output or something else.

Reply:
It's for all I/O tensors, with some rare exception that you might return a different shape (but it's dangerous to do so). I've made the title clearer.